warehouse/keyword-common/src/main/java/datawave/util/keyword/KeywordExtractor.java

```diff
@@ -99,23 +102,37 @@ public void parseOptions(Map<String,String> iteratorOptions) {
         }
     }
 
+    @Nonnull
     public KeywordResults extractKeywords() {
+        if (foundContent.isEmpty()) {
+            return EMPTY_RESULTS;
+        }
         KeywordResults results = EMPTY_RESULTS;
         for (String viewName : preferredViews) {
             for (Map.Entry<String,VisibleContent> foundEntry : foundContent.entrySet()) {
-                if (viewName.equals(foundEntry.getKey())) {
-                    VisibleContent content = foundEntry.getValue();
-                    final LinkedHashMap<String,Double> keywords = yakeKeywordExtractor.extractKeywords(content.getContent());
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Extracted {} keywords from {} view.", keywords.size(), viewName);
-                    }
-                    if (!keywords.isEmpty()) {
-                        results = new KeywordResults(source, viewName, language, content.getVisibility(), keywords);
-                        break;
+                if (viewName.endsWith("*")) {
+                    final String truncatedName = viewName.substring(0, viewName.length() - 1);
+                    if (foundEntry.getKey().startsWith(truncatedName)) {
+                        results = extractKeywordsFromVisibleContent(viewName, foundEntry.getValue());
                     }
+                } else if (viewName.equals(foundEntry.getKey())) {
+                    results = extractKeywordsFromVisibleContent(viewName, foundEntry.getValue());
+                }
+
+                if (results != EMPTY_RESULTS) {
+                    return results;
                 }
             }
         }
-        return results;
+        return EMPTY_RESULTS;
+    }
+
+    @Nonnull
```

Comment:
Consider adding this javadoc:
```suggestion
    /**
     * Attempts to extract keywords from VisibleContent object provided.
     *
     * @param viewName
     *            the view name we're extracting from.
     * @param content
     *            the content associated with that view.
     * @return a KeywordResult object containing keywords extracted from the provided content, or the canonical <code>EMPTY_RESULTS</code> KeywordResults object
     *         if no keywords could be extracted.
     */
    @Nonnull
```